alphabase/psm_reader/maxquant_reader.py

```diff
@@ -245,6 +248,20 @@ def _load_file(self, filename):
         self._find_mod_seq_column(df)
         df = df[~pd.isna(df["Retention time"])]
         df.fillna("", inplace=True)
+
+        # remove MBR PSMs as they are currently not supported and will crash import
+        mapped_columns = self._find_mapped_columns(df)
+        if "scan_num" in mapped_columns:
+            scan_num_col = mapped_columns["scan_num"]
+            no_ms2_mask = df[scan_num_col] == ""
+            if np.sum(no_ms2_mask) > 0:
+                warnings.warn(
+                    f"Maxquant psm file contains {np.sum(no_ms2_mask)} MBR PSMs without MS2 scan. This is not yet supported and rows containing MBR PSMs will be removed."
+                )
+                df = df[~no_ms2_mask]
+                df.reset_index(drop=True, inplace=True)
+                df[scan_num_col] = df[scan_num_col].astype(int)
```

Comment:
if we don't hit the `if "scan_num" in mapped_columns:` block: will the type of `scan_num_col` still be `int`?
or turning it around: we should make sure all columns' dtypes are consistent, and independent of the inner logic of the reader.